### ISSUE TITLE
Generated emit cache issue

### ISSUE THREAD

**Issue by sylc** [denoland/deno#13317]
Hi,

I'm having a deno cache update issue when `no-check` is NOT specified.
I'm trying to update a module reference but then I get `error: Loading unprepared module: <stale_module_address>`. The module address listed in the error is not present in deno info output

To reproduce:
- Clone https://github.com/sylc/issue_pin_fresh  (which is copy of https://github.com/lucacasonato/fresh/tree/main/www)
- Run `deno run -A --unstable --reload main.ts`. There is crash. `thread 'main' panicked at 'internal error: entered unreachable code: Unexpected missing emit: file:///mnt/c/Users/sylva/Documents/Projects/OpenSource/test1/issue_pin_fresh/deps.ts', cli/proc_state.rs:586:25`

<details>
  <summary>Full Stacktrace</summary>
  
  ```
  Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.17.2
Args: ["deno", "run", "-A", "--unstable", "--reload", "main.ts"]

thread 'main' panicked at 'internal error: entered unreachable code: Unexpected missing emit: file:///mnt/c/Users/sylva/Documents/Projects/OpenSource/test1/issue_pin_fresh/deps.ts', cli/proc_state.rs:586:25
stack backtrace:
   0:     0x7f8e8cdef30c - std::backtrace_rs::backtrace::libunwind::trace::hf6a6dfd7da937cb0
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x7f8e8cdef30c - std::backtrace_rs::backtrace::trace_unsynchronized::hc596a19e4891f7f3
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f8e8cdef30c - std::sys_common::backtrace::_print_fmt::hb16700db31584325
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f8e8cdef30c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h231c4190cfa75162
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x7f8e8ce19ffc - core::fmt::write::h2a1462b5f8eea807
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/fmt/mod.rs:1163:17
   5:     0x7f8e8cde5c05 - std::io::Write::write_fmt::h71ddfebc68685972
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/io/mod.rs:1696:15
   6:     0x7f8e8cdf16e0 - std::sys_common::backtrace::_print::hcc197d4bebf2b369
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x7f8e8cdf16e0 - std::sys_common::backtrace::print::h335a66af06738c7c
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x7f8e8cdf16e0 - std::panicking::default_hook::{{closure}}::h6fac9ac9c8b79e52
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:210:50
   9:     0x7f8e8cdf1295 - std::panicking::default_hook::h341c1030c6a1161b
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:227:9
  10:     0x7f8e8d1680ae - deno::setup_panic_hook::{{closure}}::h641f641302ba4689
  11:     0x7f8e8cdf1f29 - std::panicking::rust_panic_with_hook::h50680ff4b44510c6
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:628:17
  12:     0x7f8e8cdf19b0 - std::panicking::begin_panic_handler::{{closure}}::h9371c0fbb1e8465a
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:521:13
  13:     0x7f8e8cdef7b4 - std::sys_common::backtrace::__rust_end_short_backtrace::h9b3efa22a5768c0f
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/sys_common/backtrace.rs:139:18
  14:     0x7f8e8cdf1919 - rust_begin_unwind
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:517:5
  15:     0x7f8e8ce17ac1 - core::panicking::panic_fmt::h23b9203e89cc61cf
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/panicking.rs:100:14
  16:     0x7f8e8d13f295 - deno::proc_state::ProcState::load::hab14be3ab9bc3925
  17:     0x7f8e8d02218f - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h4ec687f63f789de8
  18:     0x7f8e8dab9dca - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h027602f764377892
  19:     0x7f8e8dad4f23 - <deno_core::modules::RecursiveModuleLoad as futures_core::stream::Stream>::poll_next::hebf3037627e07298
  20:     0x7f8e8d07e2a2 - <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll::hc6237c45e32c6b13
  21:     0x7f8e8d03ce95 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h8884b2b6c6efb6f1
  22:     0x7f8e8d042079 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h981464430cc71dde
  23:     0x7f8e8d162542 - deno::run_command::{{closure}}::h10534eabd0c4b78d
  24:     0x7f8e8d01b0b5 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h3251b4e4166250a7
  25:     0x7f8e8d16bf73 - deno::main::heeccf1da13807fca
  26:     0x7f8e8ceb31e8 - std::sys_common::backtrace::__rust_begin_short_backtrace::h522f3efdeb9afeb6
  27:     0x7f8e8ceb8948 - std::rt::lang_start::{{closure}}::h8a336066b14bf72f
  28:     0x7f8e8cdeefeb - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hc56adab7a77ec6e3
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/ops/function.rs:259:13
  29:     0x7f8e8cdeefeb - std::panicking::try::do_call::h29f013120c5abc65
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:403:40
  30:     0x7f8e8cdeefeb - std::panicking::try::h86d5b2b66caec4cf
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:367:19
  31:     0x7f8e8cdeefeb - std::panic::catch_unwind::h7dd136d787f51397
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panic.rs:133:14
  32:     0x7f8e8cdeefeb - std::rt::lang_start_internal::{{closure}}::h4a199351e630a8a5
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/rt.rs:128:48
  33:     0x7f8e8cdeefeb - std::panicking::try::do_call::h20ceb9e5dff838c6
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:403:40
  34:     0x7f8e8cdeefeb - std::panicking::try::hc2abb46a5e41bd43
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:367:19
  35:     0x7f8e8cdeefeb - std::panic::catch_unwind::h52c3eb4408ad6dfb
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panic.rs:133:14
  36:     0x7f8e8cdeefeb - std::rt::lang_start_internal::hd15a47be08101c28
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/rt.rs:128:20
  37:     0x7f8e8d17b34b - main
  38:     0x7f8e8a0270b3 - __libc_start_main
  39:     0x7f8e8bf8702a - _start
  40:                0x0 - unknown

  ```

</details>

- Run again with `no-check` this time `deno run -A --unstable --reload --no-check main.ts` .It is working
- Run again without `no-check` this time `deno run -A --unstable --reload main.ts`. It is still working!
- Update module address in `./deps.ts` and `./deps_server.ts` files from `https://raw.githubusercontent.com/lucacasonato/fresh/main/runtime.ts` to `https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/runtime.ts` 
- Run again `deno run -A --unstable --reload main.ts` and note the error `error: Loading unprepared module: https://raw.githubusercontent.com/lucacasonato/fresh/main/runtime.ts`
- deno info does not list any module from `../main/runtime.ts`
<details>
  <summary>deno info output</summary>
  $ deno info ./main.ts 
local: C:\Users\sylva\Documents\Projects\OpenSource\issue_pin_fresh\main.ts
emit: C:\Users\sylva\AppData\Local\deno\gen\file\C\Users\sylva\Documents\Projects\OpenSource\issue_pin_fresh\main.ts.js
type: TypeScript
dependencies: 103 unique (total 1.98MB)

file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/main.ts (106B)
├─┬ file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/deps_server.ts (251B)
│ └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/server.ts (37B)
│   └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/mod.ts (685B)
│     ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/context.ts (13.93KB)
│     │ ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/csp.ts (3.53KB)
│     │ │ └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts (419B)
│     │ │   ├─┬ https://esm.sh/preact@10.5.14 (100B)
│     │ │   │ ├─┬ https://cdn.esm.sh/v61/preact@10.5.14/src/index.d.ts (8.16KB)
│     │ │   │ │ └─┬ https://cdn.esm.sh/v61/preact@10.5.14/src/jsx.d.ts (30.96KB)
│     │ │   │ │   └── https://cdn.esm.sh/v61/preact@10.5.14/src/index.d.ts *
│     │ │   │ ├── https://cdn.esm.sh/v61/preact@10.5.14/deno/preact.js (9.8KB)
│     │ │   └─┬ https://esm.sh/preact@10.5.14/hooks (105B)
│     │ │     ├─┬ https://cdn.esm.sh/v61/preact@10.5.14/hooks/src/index.d.ts (5.37KB)
│     │ │     │ └── https://cdn.esm.sh/v61/preact@10.5.14/src/index.d.ts *
│     │ │     ├─┬ https://cdn.esm.sh/v61/preact@10.5.14/deno/hooks.js (2.63KB)
│     │ │     │ └── https://cdn.esm.sh/v61/preact@10.5.14/deno/preact.js *
│     │ ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/bundle.ts (4.34KB)
│     │ │ ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/deps.ts (1.13KB)
│     │ │ │ ├── https://crux.land/api/get/2KNRVU.ts (4.44KB)
│     │ │ │ ├─┬ https://deno.land/std@0.108.0/fs/walk.ts (5.22KB)
│     │ │ │ │ ├── https://deno.land/std@0.108.0/_util/assert.ts (405B)
│     │ │ │ │ └─┬ https://deno.land/std@0.108.0/path/mod.ts (725B)
│     │ │ │ │   ├── https://deno.land/std@0.108.0/_util/os.ts (598B)
│     │ │ │ │   ├── https://deno.land/std@0.108.0/path/_interface.ts (728B)
│     │ │ │ │   ├─┬ https://deno.land/std@0.108.0/path/common.ts (1.16KB)
│     │ │ │ │   │ └─┬ https://deno.land/std@0.108.0/path/separator.ts (259B)
│     │ │ │ │   │   └── https://deno.land/std@0.108.0/_util/os.ts *
│     │ │ │ │   ├─┬ https://deno.land/std@0.108.0/path/glob.ts (12.36KB)
│     │ │ │ │   │ ├── https://deno.land/std@0.108.0/_util/os.ts *
│     │ │ │ │   │ ├─┬ https://deno.land/std@0.108.0/path/posix.ts (14.49KB)
│     │ │ │ │   │ │ ├── https://deno.land/std@0.108.0/path/_constants.ts (1.9KB)
│     │ │ │ │   │ │ ├── https://deno.land/std@0.108.0/path/_interface.ts *
│     │ │ │ │   │ │ └─┬ https://deno.land/std@0.108.0/path/_util.ts (3.62KB)
│     │ │ │ │   │ │   ├── https://deno.land/std@0.108.0/path/_constants.ts *
│     │ │ │ │   │ │   └── https://deno.land/std@0.108.0/path/_interface.ts *
│     │ │ │ │   │ ├── https://deno.land/std@0.108.0/path/separator.ts *
│     │ │ │ │   │ └─┬ https://deno.land/std@0.108.0/path/win32.ts (29.41KB)
│     │ │ │ │   │   ├── https://deno.land/std@0.108.0/_util/assert.ts *
│     │ │ │ │   │   ├── https://deno.land/std@0.108.0/path/_constants.ts *
│     │ │ │ │   │   ├── https://deno.land/std@0.108.0/path/_interface.ts *
│     │ │ │ │   │   └── https://deno.land/std@0.108.0/path/_util.ts *
│     │ │ │ │   ├── https://deno.land/std@0.108.0/path/posix.ts *
│     │ │ │ │   ├── https://deno.land/std@0.108.0/path/separator.ts *
│     │ │ │ │   └── https://deno.land/std@0.108.0/path/win32.ts *
│     │ │ │ ├─┬ https://deno.land/std@0.108.0/http/server.ts (18.12KB)
│     │ │ │ │ └─┬ https://deno.land/std@0.108.0/async/mod.ts (290B)
│     │ │ │ │   ├─┬ https://deno.land/std@0.108.0/async/deadline.ts (504B)
│     │ │ │ │   │ └── https://deno.land/std@0.108.0/async/deferred.ts (1.45KB)
│     │ │ │ │   ├── https://deno.land/std@0.108.0/async/debounce.ts (2.04KB)
│     │ │ │ │   ├── https://deno.land/std@0.108.0/async/deferred.ts *
│     │ │ │ │   ├── https://deno.land/std@0.108.0/async/delay.ts (679B)
│     │ │ │ │   ├─┬ https://deno.land/std@0.108.0/async/mux_async_iterator.ts (1.94KB)
│     │ │ │ │   │ └── https://deno.land/std@0.108.0/async/deferred.ts *
│     │ │ │ │   ├── https://deno.land/std@0.108.0/async/pool.ts (2.58KB)
│     │ │ │ │   └── https://deno.land/std@0.108.0/async/tee.ts (2.57KB)
│     │ │ │ ├── https://deno.land/std@0.108.0/path/mod.ts *
│     │ │ │ ├─┬ https://deno.land/x/esbuild@v0.13.2/mod.js (62.31KB)
│     │ │ │ │ ├── https://deno.land/x/esbuild@v0.13.2/mod.d.ts (10.96KB)
│     │ │ │ │ ├─┬ https://deno.land/x/denoflate@1.2.1/mod.ts (206B)
│     │ │ │ │ │ ├── https://deno.land/x/denoflate@1.2.1/pkg/denoflate.js (7.44KB)
│     │ │ │ │ │ └── https://deno.land/x/denoflate@1.2.1/pkg/denoflate_bg.wasm.js (291.78KB)
│     │ │ │ ├─┬ https://deno.land/x/esbuild_deno_loader@0.3.0/mod.ts (2.69KB)
│     │ │ │ │ ├─┬ https://deno.land/x/esbuild_deno_loader@0.3.0/deps.ts (467B)
│     │ │ │ │ │ ├─┬ https://deno.land/std@0.95.0/path/mod.ts (725B)
│     │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/_util/os.ts (456B)
│     │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/_interface.ts (728B)
│     │ │ │ │ │ │ ├─┬ https://deno.land/std@0.95.0/path/common.ts (1.14KB)
│     │ │ │ │ │ │ │ └─┬ https://deno.land/std@0.95.0/path/separator.ts (259B)
│     │ │ │ │ │ │ │   └── https://deno.land/std@0.95.0/_util/os.ts *
│     │ │ │ │ │ │ ├─┬ https://deno.land/std@0.95.0/path/glob.ts (12.28KB)
│     │ │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/_util/os.ts *
│     │ │ │ │ │ │ │ ├─┬ https://deno.land/std@0.95.0/path/posix.ts (14.28KB)
│     │ │ │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/_constants.ts (1.9KB)
│     │ │ │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/_interface.ts *
│     │ │ │ │ │ │ │ │ └─┬ https://deno.land/std@0.95.0/path/_util.ts (3.62KB)
│     │ │ │ │ │ │ │ │   ├── https://deno.land/std@0.95.0/path/_constants.ts *
│     │ │ │ │ │ │ │ │   └── https://deno.land/std@0.95.0/path/_interface.ts *
│     │ │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/separator.ts *
│     │ │ │ │ │ │ │ └─┬ https://deno.land/std@0.95.0/path/win32.ts (29.13KB)
│     │ │ │ │ │ │ │   ├── https://deno.land/std@0.95.0/_util/assert.ts (405B)
│     │ │ │ │ │ │ │   ├── https://deno.land/std@0.95.0/path/_constants.ts *
│     │ │ │ │ │ │ │   ├── https://deno.land/std@0.95.0/path/_interface.ts *
│     │ │ │ │ │ │ │   └── https://deno.land/std@0.95.0/path/_util.ts *
│     │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/posix.ts *
│     │ │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/separator.ts *
│     │ │ │ │ │ │ └── https://deno.land/std@0.95.0/path/win32.ts *
│     │ │ │ │ │ ├── https://deno.land/std@0.95.0/path/posix.ts *
│     │ │ │ │ │ ├── https://deno.land/x/esbuild@v0.12.9/mod.d.ts (10.3KB)
│     │ │ │ │ │ └─┬ https://deno.land/x/importmap@0.2.0/mod.ts (7.94KB)
│     │ │ │ │ │   └── https://deno.land/x/importmap@0.2.0/_util.ts (1.46KB)
│     │ │ │ │ ├── https://deno.land/x/esbuild_deno_loader@0.3.0/src/deno.ts (1.37KB)
│     │ │ │ │ ├─┬ https://deno.land/x/esbuild_deno_loader@0.3.0/src/native_loader.ts (1.67KB)
│     │ │ │ │ │ ├── https://deno.land/x/esbuild_deno_loader@0.3.0/deps.ts *
│     │ │ │ │ │ └── https://deno.land/x/esbuild_deno_loader@0.3.0/src/deno.ts *
│     │ │ │ │ └─┬ https://deno.land/x/esbuild_deno_loader@0.3.0/src/portable_loader.ts (4.12KB)
│     │ │ │ │   ├── https://deno.land/x/esbuild_deno_loader@0.3.0/deps.ts *
│     │ │ │ │   └── https://deno.land/x/esbuild_deno_loader@0.3.0/src/deno.ts *
│     │ │ │ ├─┬ https://deno.land/x/media_types@v2.10.2/mod.ts (4.22KB)
│     │ │ │ │ ├── https://deno.land/x/media_types@v2.10.2/db.ts (180.79KB)
│     │ │ │ │ └─┬ https://deno.land/x/media_types@v2.10.2/deps.ts (139B)
│     │ │ │ │   └─┬ https://deno.land/std@0.107.0/path/mod.ts (725B)
│     │ │ │ │     ├── https://deno.land/std@0.107.0/_util/os.ts (598B)
│     │ │ │ │     ├── https://deno.land/std@0.107.0/path/_interface.ts (728B)
│     │ │ │ │     ├─┬ https://deno.land/std@0.107.0/path/common.ts (1.16KB)
│     │ │ │ │     │ └─┬ https://deno.land/std@0.107.0/path/separator.ts (259B)
│     │ │ │ │     │   └── https://deno.land/std@0.107.0/_util/os.ts *
│     │ │ │ │     ├─┬ https://deno.land/std@0.107.0/path/glob.ts (12.36KB)
│     │ │ │ │     │ ├── https://deno.land/std@0.107.0/_util/os.ts *
│     │ │ │ │     │ ├─┬ https://deno.land/std@0.107.0/path/posix.ts (14.49KB)
│     │ │ │ │     │ │ ├── https://deno.land/std@0.107.0/path/_constants.ts (1.9KB)
│     │ │ │ │     │ │ ├── https://deno.land/std@0.107.0/path/_interface.ts *
│     │ │ │ │     │ │ └─┬ https://deno.land/std@0.107.0/path/_util.ts (3.62KB)
│     │ │ │ │     │ │   ├── https://deno.land/std@0.107.0/path/_constants.ts *
│     │ │ │ │     │ │   └── https://deno.land/std@0.107.0/path/_interface.ts *
│     │ │ │ │     │ ├── https://deno.land/std@0.107.0/path/separator.ts *
│     │ │ │ │     │ └─┬ https://deno.land/std@0.107.0/path/win32.ts (29.41KB)
│     │ │ │ │     │   ├── https://deno.land/std@0.107.0/_util/assert.ts (405B)
│     │ │ │ │     │   ├── https://deno.land/std@0.107.0/path/_constants.ts *
│     │ │ │ │     │   ├── https://deno.land/std@0.107.0/path/_interface.ts *
│     │ │ │ │     │   └── https://deno.land/std@0.107.0/path/_util.ts *
│     │ │ │ │     ├── https://deno.land/std@0.107.0/path/posix.ts *
│     │ │ │ │     ├── https://deno.land/std@0.107.0/path/separator.ts *
│     │ │ │ │     └── https://deno.land/std@0.107.0/path/win32.ts *
│     │ │ │ ├─┬ https://esm.sh/preact-render-to-string@5.1.19?deps=preact@10.5.14 (320B)
│     │ │ │ │ ├─┬ https://cdn.esm.sh/v61/preact-render-to-string@5.1.19/src/index.d.ts (429B)
│     │ │ │ │ │ └─┬ https://cdn.esm.sh/v61/preact@10.6.4/src/index.d.ts (8.16KB)
│     │ │ │ │ │   └─┬ https://cdn.esm.sh/v61/preact@10.6.4/src/jsx.d.ts (31.77KB)
│     │ │ │ │ │     └── https://cdn.esm.sh/v61/preact@10.6.4/src/index.d.ts *
│     │ │ │ │ ├─┬ https://cdn.esm.sh/v61/preact-render-to-string@5.1.19/X-ZGVwczpwcmVhY3RAMTAuNS4xNA/deno/preact-render-to-string.js (4.51KB)
│     │ │ │ │ │ └── https://cdn.esm.sh/v61/preact@10.5.14/deno/preact.js *
│     │ │ │ ├── https://gist.githubusercontent.com/lucacasonato/358c6b7e8198bfb2cf3d220e49fdcf5f/raw/3714cb0f59606eefc29ed0fea36d4cd93549938b/esbuild-wasm.js (130.8KB)
│     │ │ │ └── https://unpkg.com/esbuild-wasm@0.11.19/esm/browser.d.ts (10.01KB)
│     │ │ └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/types.ts (1.21KB)
│     │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts *
│     │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/types.ts (1.76KB)
│     │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/deps.ts *
│     │ │   └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/render.tsx (10.84KB)
│     │ │     ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/csp.ts *
│     │ │     ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts *
│     │ │     ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/head.ts (540B)
│     │ │     │ └── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts *
│     │ │     ├─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/hooks.ts (2.16KB)
│     │ │     │ ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts *
│     │ ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/render.tsx *
│     │ └── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/types.ts *
│     ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/deps.ts *
│     ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/render.tsx *
│     └── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/server/types.ts *
└─┬ file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/routes.gen.ts (403B)
  ├─┬ file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/pages/_render.ts (606B)
  │ ├─┬ file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/deps.ts (911B)
  │ │ ├─┬ https://esm.sh/twind (97B)
  │ │ │ ├─┬ https://cdn.esm.sh/v61/twind@0.16.16/twind.d.ts (28.98KB)
  │ │ │ │ └── https://cdn.esm.sh/v61/csstype@3.0.10/index.d.ts (844.54KB)
  │ │ │ ├─┬ https://cdn.esm.sh/v61/twind@0.16.16/deno/twind.js (32.54KB)
  │ │ │ │ └── https://cdn.esm.sh/v61/style-vendorizer@2.1.1/deno/style-vendorizer.js (1.77KB)
  │ │ └─┬ https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/runtime.ts (279B)
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/csp.ts *
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/deps.ts *
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/head.ts *
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/hooks.ts *
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/suspense.ts *
  │ │   ├── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/types.ts *
  │ │   └── https://raw.githubusercontent.com/lucacasonato/fresh/6abf62c832ced41ae9365b48939221adfbe74e5d/src/runtime/utils.ts *
  │ ├── file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/deps_server.ts *
  │ └─┬ https://esm.sh/twind/sheets (105B)
  │   ├─┬ https://cdn.esm.sh/v61/twind@0.16.16/sheets/sheets.d.ts (1.77KB)
  │   │ └── https://cdn.esm.sh/v61/twind@0.16.16/twind.d.ts *
  │   ├── https://cdn.esm.sh/v61/twind@0.16.16/deno/sheets.js (988B)
  └─┬ file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/pages/index.tsx (4.41KB)
    └── file:///C:/Users/sylva/Documents/Projects/OpenSource/issue_pin_fresh/deps.ts *
</details>

- In the deno/gen folder, `deps.ts.js` still refer to `.../fresh/main/runtime.ts`
- with no-check, it works `deno run -A --unstable --reload --no-check main.ts`
- after no-check has run, the `deps.ts.js` is now referring correctly to the `.../fresh/main/runtime.ts`

$ deno --version
deno 1.17.2 (release, x86_64-pc-windows-msvc)
v8 9.7.106.15
typescript 4.5.2

thanks

**Comment by GJZwiers** [comment]
I think this issue is the same as #14093, because the `deps.ts` uses `declare global` and also `no-default-lib="true"`, which leads to a bug in TypeScript. The bug has been fixed upstream, so I think this should be fixed when the TypeScript fix lands in Deno. @dsherret what do you think?

**Comment by nayeemrmn** [comment]
Fixed in #15022
